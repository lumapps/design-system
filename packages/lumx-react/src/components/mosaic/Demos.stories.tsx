export default { title: 'LumX components/mosaic/Demos' };
