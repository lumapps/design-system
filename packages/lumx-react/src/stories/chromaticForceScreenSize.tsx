/* eslint-disable */
import React from 'react';
import isChromatic from 'chromatic/isChromatic';

/** Story decorator used to force a minimum screen size when running in chromatic */
export const chromaticForceScreenSize = (story: any) =>
    isChromatic() ? <div style={{ minWidth: 1200, minHeight: 800 }}>{story()}</div> : story();
