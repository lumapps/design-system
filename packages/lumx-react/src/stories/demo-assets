../../../site-demo/static/demo-assets