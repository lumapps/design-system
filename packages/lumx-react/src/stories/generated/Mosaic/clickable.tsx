../../../../../site-demo/content/product/components/mosaic/react/clickable.tsx