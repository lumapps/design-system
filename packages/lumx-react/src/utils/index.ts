/**
 * WARNING: All modules exported here are exposed to NPM in '@lumx/react/utils'.
 */

export { ClickAwayProvider } from './ClickAwayProvider';
export { Portal, type PortalProps, type PortalInit, PortalProvider, type PortalProviderProps } from './Portal';
