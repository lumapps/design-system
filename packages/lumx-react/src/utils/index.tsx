export * from '@lumx/core/js/utils';

export * from './getRootClassName';

export * from './htmlDecode';

export * from './partitionMulti';

export * from './type';
