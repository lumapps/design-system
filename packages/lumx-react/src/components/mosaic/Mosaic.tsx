import React, { forwardRef, MouseEventHandler, useMemo } from 'react';

import { Alignment, AspectRatio, Theme, Thumbnail, ThumbnailProps } from '@lumx/react';
import { Comp, GenericProps, getRootClassName, handleBasicClasses } from '@lumx/react/utils';
import classNames from 'classnames';
import take from 'lodash/take';

/**
 * Defines the props of the component.
 */
export interface MosaicProps extends GenericProps {
    /** Theme adapting the component to light or dark background. */
    theme?: Theme;
    /** Thumbnails. */
    thumbnails: ThumbnailProps[];
    /** On image click callback. */
    onImageClick?(index: number): void;
}

/**
 * Component display name.
 */
const COMPONENT_NAME = 'Mosaic';

/**
 * Component default class name and class prefix.
 */
const CLASSNAME = getRootClassName(COMPONENT_NAME);

/**
 * Component default props.
 */
const DEFAULT_PROPS: Partial<MosaicProps> = {
    theme: Theme.light,
};

/**
 * Mosaic component.
 *
 * @param  props Component props.
 * @param  ref   Component ref.
 * @return React element.
 */
export const Mosaic: Comp<MosaicProps, HTMLDivElement> = forwardRef((props, ref) => {
    const { className, theme, thumbnails, onImageClick, ...forwardedProps } = props;
    const handleImageClick = useMemo(() => {
        if (!onImageClick) return undefined;

        return (index: number, onClick?: MouseEventHandler): MouseEventHandler => (event) => {
            onClick?.(event);
            onImageClick?.(index);
        };
    }, [onImageClick]);

    return (
        <div
            ref={ref}
            {...forwardedProps}
            className={classNames(className, handleBasicClasses({ prefix: CLASSNAME, theme }), {
                [`${CLASSNAME}--has-1-thumbnail`]: thumbnails?.length === 1,
                [`${CLASSNAME}--has-2-thumbnails`]: thumbnails?.length === 2,
                [`${CLASSNAME}--has-3-thumbnails`]: thumbnails?.length === 3,
                [`${CLASSNAME}--has-4-thumbnails`]: thumbnails?.length >= 4,
            })}
        >
            <div className={`${CLASSNAME}__wrapper`}>
                {take(thumbnails, 4).map((thumbnail: any, index: number) => {
                    const { image, onClick, align, ...thumbnailProps } = thumbnail;

                    return (
                        <div key={index} className={`${CLASSNAME}__thumbnail`}>
                            <Thumbnail
                                {...thumbnailProps}
                                align={align || Alignment.left}
                                image={image}
                                theme={theme}
                                aspectRatio={AspectRatio.free}
                                fillHeight
                                onClick={handleImageClick?.(index, onClick) || onClick}
                            />

                            {thumbnails.length > 4 && index === 3 && (
                                <div className={`${CLASSNAME}__overlay`}>
                                    <span>+{thumbnails.length - 4}</span>
                                </div>
                            )}
                        </div>
                    );
                })}
            </div>
        </div>
    );
});
Mosaic.displayName = COMPONENT_NAME;
Mosaic.className = CLASSNAME;
Mosaic.defaultProps = DEFAULT_PROPS;
