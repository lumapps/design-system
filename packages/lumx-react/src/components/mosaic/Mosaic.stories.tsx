import React, { useCallback, useRef, useState } from 'react';

import { Alignment, ImageBlock, Lightbox, Slideshow, SlideshowItem, Theme } from '@lumx/react';
import { boolean, number } from '@storybook/addon-knobs';
import { thumbnailsKnob } from '@lumx/react/stories/knobs/thumbnailsKnob';
import { Mosaic } from './Mosaic';

export default { title: 'LumX components/mosaic/Mosaic' };

export const OneThumbnail = ({ theme }: any) => (
    <div style={{ width: 250 }}>
        <Mosaic theme={theme} thumbnails={thumbnailsKnob(1)} />
    </div>
);

export const TwoThumbnails = ({ theme }: any) => (
    <div style={{ width: 250 }}>
        <Mosaic theme={theme} thumbnails={thumbnailsKnob(2)} />
    </div>
);

export const ThreeThumbnails = ({ theme }: any) => (
    <div style={{ width: 250 }}>
        <Mosaic theme={theme} thumbnails={thumbnailsKnob(3)} />
    </div>
);

export const FourThumbnails = ({ theme }: any) => (
    <div style={{ width: 250 }}>
        <Mosaic theme={theme} thumbnails={thumbnailsKnob(4)} />
    </div>
);

export const FiveThumbnails = ({ theme }: any) => (
    <div style={{ width: 250 }}>
        <Mosaic theme={theme} thumbnails={thumbnailsKnob(5)} />
    </div>
);

export const SixThumbnails = ({ theme }: any) => {
    const enableSlideShow = boolean('Enable slideshow', true);
    const thumbnails = thumbnailsKnob(number('Number of thumbnails', 6, { min: 1, max: 6 }));
    const [activeIndex, setActiveIndex] = useState<number>();
    const lightBoxParent = useRef(null);
    const closeLightBox = useCallback(() => {
        setActiveIndex(undefined);
    }, [setActiveIndex]);

    return (
        <div ref={lightBoxParent} style={{ width: 250 }}>
            <Mosaic theme={theme} onImageClick={enableSlideShow ? setActiveIndex : undefined} thumbnails={thumbnails} />

            {enableSlideShow && (
                <Lightbox
                    isOpen={activeIndex !== undefined}
                    parentElement={lightBoxParent}
                    onClose={closeLightBox}
                    closeButtonProps={{ label: 'Close' }}
                >
                    <Slideshow
                        activeIndex={activeIndex}
                        slideshowControlsProps={{
                            nextButtonProps: { label: 'Next' },
                            previousButtonProps: { label: 'Previous' },
                        }}
                        fillHeight
                        theme={Theme.dark}
                    >
                        {thumbnails.map((thumbnail, index) => (
                            <SlideshowItem key={`${thumbnail.alt}-${thumbnail.image}-${index}`}>
                                <ImageBlock
                                    alt={thumbnail.alt}
                                    image={thumbnail.image}
                                    theme={Theme.dark}
                                    align={Alignment.center}
                                    fillHeight
                                />
                            </SlideshowItem>
                        ))}
                    </Slideshow>
                </Lightbox>
            )}
        </div>
    );
};
