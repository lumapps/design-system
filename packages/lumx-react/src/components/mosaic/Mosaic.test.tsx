import { Mosaic, MosaicProps } from '@lumx/react/components/mosaic/Mosaic';
import { commonTestsSuite, Wrapper } from '@lumx/react/testing/utils';

import { mount, shallow } from 'enzyme';
import 'jest-enzyme';

import React, { ReactElement } from 'react';
import { Theme } from '..';
import * as stories from './Mosaic.stories';

const CLASSNAME = Mosaic.className as string;

// Mock out the useIntersectionObserver hook since it can't work with Jest/Enzyme.
jest.mock('@lumx/react/hooks/useIntersectionObserver', () => ({
    useIntersectionObserver: () => new Map(),
}));

type SetupProps = Partial<MosaicProps>;

/**
 * Mounts the component and returns common DOM elements / data needed in multiple tests further down.
 */
const setup = (propsOverride: SetupProps = {}, shallowRendering = true) => {
    const props: any = { ...propsOverride };
    const renderer: (el: ReactElement) => Wrapper = shallowRendering ? shallow : mount;
    const wrapper = renderer(<Mosaic {...props} />);

    return {
        thumbnails: wrapper.find('Thumbnail'),
        props,
        wrapper,
    };
};

describe(`<${Mosaic.displayName}>`, () => {
    // 1. Test render via snapshot.
    describe('Snapshots and structure', () => {
        // Do snapshot render test on every stories.
        for (const [storyName, Story] of Object.entries(stories)) {
            if (typeof Story !== 'function') {
                continue;
            }

            it(`should render story ${storyName}`, () => {
                const wrapper = shallow(<Story />);
                expect(wrapper.find('Mosaic').dive()).toMatchSnapshot();
            });
        }
    });

    // 2. Test defaultProps value and important props custom values.
    describe('Props', () => {
        it('should pass theme prop to Thumbnails', () => {
            const expectedTheme = Theme.dark;
            const { thumbnails } = setup({
                theme: expectedTheme,
                thumbnails: [
                    { alt: 'image0', image: 'image/file/path/0' },
                    { alt: 'image1', image: 'image/file/path/1' },
                    { alt: 'image2', image: 'image/file/path/2' },
                    { alt: 'image3', image: 'image/file/path/3' },
                ],
            });
            thumbnails.forEach((thumbnail: Wrapper) => {
                expect(thumbnail.prop('theme')).toBe(expectedTheme);
            });
        });
    });

    // 3. Test events.
    describe('Events', () => {
        it('should keep Thumbnail onClick', () => {
            const onClick = jest.fn();
            const { thumbnails } = setup({
                thumbnails: [
                    { alt: 'image0', image: 'image/file/path/0', onClick },
                    { alt: 'image1', image: 'image/file/path/1' },
                    { alt: 'image2', image: 'image/file/path/2' },
                    { alt: 'image3', image: 'image/file/path/3' },
                ],
            });
            thumbnails.forEach((thumbnail: Wrapper) => {
                thumbnail.simulate('click');
            });
            expect(onClick).toHaveBeenCalledTimes(1);
        });

        it('should handle both Thumbnail onClick and Mosaic onImageClick', () => {
            const onImageClick = jest.fn();
            const onClick = jest.fn();
            const { thumbnails } = setup({
                onImageClick,
                thumbnails: [
                    { alt: 'image0', image: 'image/file/path/0', onClick },
                    { alt: 'image1', image: 'image/file/path/1' },
                    { alt: 'image2', image: 'image/file/path/2' },
                    { alt: 'image3', image: 'image/file/path/3' },
                ],
            });
            thumbnails.forEach((thumbnail: Wrapper) => {
                thumbnail.simulate('click');
            });
            expect(onClick).toHaveBeenCalledTimes(1);
            expect(onImageClick).toHaveBeenCalledTimes(4);
        });
    });

    // 4. Test conditions (i.e. things that display or not in the UI based on props).
    describe('Conditions', () => {
        // Nothing to do here.
    });

    // 5. Test state.
    describe('State', () => {
        // Nothing to do here.
    });

    // Common tests suite.
    commonTestsSuite(setup, { className: 'wrapper' }, { className: CLASSNAME });
});
