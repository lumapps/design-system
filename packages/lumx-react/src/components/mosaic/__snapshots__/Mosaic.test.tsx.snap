// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Mosaic> Snapshots and structure should render story FiveThumbnails 1`] = `
<div
  className="lumx-mosaic lumx-mosaic--theme-light lumx-mosaic--has-4-thumbnails"
>
  <div
    className="lumx-mosaic__wrapper"
  >
    <div
      className="lumx-mosaic__thumbnail"
      key="0"
    >
      <Thumbnail
        align="left"
        alt="Image 1"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape1.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="1"
    >
      <Thumbnail
        align="left"
        alt="Image 2"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape2.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="2"
    >
      <Thumbnail
        align="left"
        alt="Image 3"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape3.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="3"
    >
      <Thumbnail
        align="left"
        alt="Image 4"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="portrait1.jpg"
        loading="lazy"
        theme="light"
      />
      <div
        className="lumx-mosaic__overlay"
      >
        <span>
          +
          1
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Mosaic> Snapshots and structure should render story FourThumbnails 1`] = `
<div
  className="lumx-mosaic lumx-mosaic--theme-light lumx-mosaic--has-4-thumbnails"
>
  <div
    className="lumx-mosaic__wrapper"
  >
    <div
      className="lumx-mosaic__thumbnail"
      key="0"
    >
      <Thumbnail
        align="left"
        alt="Image 1"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape1.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="1"
    >
      <Thumbnail
        align="left"
        alt="Image 2"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape2.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="2"
    >
      <Thumbnail
        align="left"
        alt="Image 3"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape3.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="3"
    >
      <Thumbnail
        align="left"
        alt="Image 4"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="portrait1.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
  </div>
</div>
`;

exports[`<Mosaic> Snapshots and structure should render story OneThumbnail 1`] = `
<div
  className="lumx-mosaic lumx-mosaic--theme-light lumx-mosaic--has-1-thumbnail"
>
  <div
    className="lumx-mosaic__wrapper"
  >
    <div
      className="lumx-mosaic__thumbnail"
      key="0"
    >
      <Thumbnail
        align="left"
        alt="Image 1"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape1.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
  </div>
</div>
`;

exports[`<Mosaic> Snapshots and structure should render story SixThumbnails 1`] = `
<div
  className="lumx-mosaic lumx-mosaic--theme-light lumx-mosaic--has-4-thumbnails"
>
  <div
    className="lumx-mosaic__wrapper"
  >
    <div
      className="lumx-mosaic__thumbnail"
      key="0"
    >
      <Thumbnail
        align="left"
        alt="Image 1"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape1.jpg"
        loading="lazy"
        onClick={[Function]}
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="1"
    >
      <Thumbnail
        align="left"
        alt="Image 2"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape2.jpg"
        loading="lazy"
        onClick={[Function]}
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="2"
    >
      <Thumbnail
        align="left"
        alt="Image 3"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape3.jpg"
        loading="lazy"
        onClick={[Function]}
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="3"
    >
      <Thumbnail
        align="left"
        alt="Image 4"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="portrait1.jpg"
        loading="lazy"
        onClick={[Function]}
        theme="light"
      />
      <div
        className="lumx-mosaic__overlay"
      >
        <span>
          +
          2
        </span>
      </div>
    </div>
  </div>
</div>
`;

exports[`<Mosaic> Snapshots and structure should render story ThreeThumbnails 1`] = `
<div
  className="lumx-mosaic lumx-mosaic--theme-light lumx-mosaic--has-3-thumbnails"
>
  <div
    className="lumx-mosaic__wrapper"
  >
    <div
      className="lumx-mosaic__thumbnail"
      key="0"
    >
      <Thumbnail
        align="left"
        alt="Image 1"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape1.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="1"
    >
      <Thumbnail
        align="left"
        alt="Image 2"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape2.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="2"
    >
      <Thumbnail
        align="left"
        alt="Image 3"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape3.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
  </div>
</div>
`;

exports[`<Mosaic> Snapshots and structure should render story TwoThumbnails 1`] = `
<div
  className="lumx-mosaic lumx-mosaic--theme-light lumx-mosaic--has-2-thumbnails"
>
  <div
    className="lumx-mosaic__wrapper"
  >
    <div
      className="lumx-mosaic__thumbnail"
      key="0"
    >
      <Thumbnail
        align="left"
        alt="Image 1"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape1.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
    <div
      className="lumx-mosaic__thumbnail"
      key="1"
    >
      <Thumbnail
        align="left"
        alt="Image 2"
        aspectRatio="free"
        fallback="M21,5V11.59L18,8.58L14,12.59L10,8.59L6,12.59L3,9.58V5A2,2 0 0,1 5,3H19A2,2 0 0,1 21,5M18,11.42L21,14.43V19A2,2 0 0,1 19,21H5A2,2 0 0,1 3,19V12.42L6,15.41L10,11.41L14,15.41"
        fillHeight={true}
        image="landscape2.jpg"
        loading="lazy"
        theme="light"
      />
    </div>
  </div>
</div>
`;
